src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,55 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    private final Machine truck = new Truck();
+    private final Machine excavator = new Excavator();
+    private final Machine bulldozer = new Bulldozer();
```

Comment:
Consider using specific type parameters instead of the raw Machine type for the fields 'truck', 'excavator', and 'bulldozer'. This will ensure type safety.